### DISCUSSION TITLE
Strive to keep diagram aspect ratio more square

### DISCUSSION THREAD

**Discussion by vedeja** [structurizr/ui#58] [Ideas]
As diagrams grow, they tend to grow in only one axis. In time, they are so horizontally stretched that they cannot be read. It would be great it the layout could strive to keep the diagram more square (or perhaps golden ratio) to make the best use of screen real estate.

Example:
![image](https://github.com/structurizr/ui/assets/15958697/7b5f5274-19f2-4a9b-90f9-ee0d26ea2ec8)

**Comment by simonbrowndotje** [comment]
This is a consequence of using automatic layout I'm afraid. You're welcome to submit a PR if you have some ideas on how it can be improved.

**Reply by vedeja** [reply]
Would such changes happen in structurizr/ui or elsewhere?

**Reply by simonbrowndotje** [reply]
`structurizr/ui` or the [structurizr-autolayout](https://github.com/structurizr/java/tree/master/structurizr-autolayout) library if you plan to do something like this -> https://stackoverflow.com/questions/11134946/distribute-nodes-on-the-same-rank-of-a-wide-graph-to-different-lines (I don't know whether this works with groups/subgraphs).
